browser/brave_stats_updater_params.cc

```diff
@@ -119,10 +138,20 @@ base::Time BraveStatsUpdaterParams::GetCurrentTimeNow() const {
   return g_current_time.is_null() ? base::Time::Now() : g_current_time;
 }
 
+// static
+bool BraveStatsUpdaterParams::GetFirstRun() const {
+  return g_first_run || first_run::IsChromeFirstRun();
```

Comment:
Is `IsChromeFirstRun` implemented on Android? I seem to recall it was implemented in Java on Android, but maybe I'm wrong. Maybe this could potentially be refactored to call/use `BraveReferralsService::GetFirstRunTime`? Not sure if that would be too complicated.

Reply:
I considered using the referrals code. On my first pass I added this on top of my other PR and moved that code to be shared. Since the referrals code runs ‘fstat’ on the sentinel file it can block, needs to go on a separate thread, and things just get messy. 

And it looks like everything in ‘first_run::’ uses the sentinel file. So we’ll need something else for Android.

Reply:
Any thoughts here?

Reply:
It's been a while, but I believe we set a first run timestamp preference on Android in order to work around this. If you're literally just trying to figure out if it's a first run, maybe on Android you could just check if that preference has a time in it?